cvat/apps/engine/views.py

```diff
@@ -685,39 +685,75 @@ def jobs(self, request, pk):
 
         return Response(serializer.data)
 
-    def upload_finished(self, request):
+    # UploadMixin method
+    def can_upload(self, data_type):
```

Comment:
We have a check `elif task_data.size != 0:` inside `data` action. Why do we need the extra check here? Also it will check permissions one more time due to call of self.get_object(). Is it really necessary?

Reply:
It is some outdated check, removed it